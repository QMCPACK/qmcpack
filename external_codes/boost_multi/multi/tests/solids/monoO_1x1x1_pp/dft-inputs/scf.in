&CONTROL
   calculation     = 'scf'
   disk_io         = 'low'
   outdir          = 'pwscf_output'
   prefix          = 'pwscf'
   pseudo_dir      = './'
   restart_mode    = 'from_scratch'
   tprnfor         = .false.
   tstress         = .false.
   verbosity       = 'high'
   wf_collect      = .true.
/

&SYSTEM
   celldm(1)       = 1.0
   degauss         = 0.0001
   ecutrho         = 1000
   ecutwfc         = 250
   ibrav           = 0
   input_dft       = 'lda'
   nat             = 2
   nosym           = .true.
   ntyp            = 1
   occupations     = 'smearing'
   smearing        = 'fermi-dirac'
   tot_charge      = 0
/

&ELECTRONS
   conv_thr        = 1e-08
   electron_maxstep = 1000
   mixing_beta     = 0.7
/


ATOMIC_SPECIES 
   O  15.999 O.BFD.upf

ATOMIC_POSITIONS alat
   O       -0.00000000      -0.00000000       1.08659253 
   O        0.00000000       0.00000000      -1.08659253 

K_POINTS automatic
   1 1 1  0 0 0 

CELL_PARAMETERS cubic
         5.10509515      -3.23993545      -0.00000000 
         5.10509515       3.23993545       0.00000000 
        -6.49690625      -0.00000000       7.08268015 


