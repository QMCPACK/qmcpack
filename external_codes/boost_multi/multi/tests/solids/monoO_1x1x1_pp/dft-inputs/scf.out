
     Program PWSCF v.4.2        starts on 31Oct2014 at 13:24:43 

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please acknowledge
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
          URL http://www.quantum-espresso.org", 
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/wiki/index.php/Citing_Quantum-ESPRESSO

     Parallel version (MPI), running on    16 processors
     R & G space division:  proc/pool =   16

     EXPERIMENTAL VERSION WITH EXACT EXCHANGE

     Current dimensions of program PWSCF are:
     Max number of different atomic species (ntypx) = 10
     Max number of k-points (npk) =  40000
     Max angular momentum in pseudopotentials (lmaxx) =  3
     Waiting for input...

     XC functional enforced from input :
     Exchange-correlation      = LDA (1100)
     EXX-fraction              =        0.00
     !!! Any further DFT definition will be discarded
     !!! Please, verify this is what you really want !

     file O.BFD.upf: wavefunction(s)  0s 0p renormalized

     Subspace diagonalization in iterative solution of the eigenvalue problem:
     Too few electrons for parallel  algorithm: we need # of bands >= SQRT(nproc)
     a serial algorithm will be used


     Planes per process (thick) : nr3 = 100 npp =    7 ncplane =  4096
 
     Proc/  planes cols     G    planes cols    G      columns  G
     Pool       (dense grid)       (smooth grid)      (wavefct grid)
        1     7    165     7827    7    165     7827     41      978
        2     7    165     7827    7    165     7827     41      978
        3     7    165     7827    7    165     7827     41      978
        4     7    165     7826    7    165     7826     41      978
        5     6    165     7826    6    165     7826     41      978
        6     6    164     7822    6    164     7822     41      978
        7     6    164     7822    6    164     7822     41      978
        8     6    164     7822    6    164     7822     41      978
        9     6    165     7826    6    165     7826     41      978
       10     6    165     7826    6    165     7826     41      978
       11     6    165     7824    6    165     7824     42      981
       12     6    165     7824    6    165     7824     41      978
       13     6    165     7824    6    165     7824     42      981
       14     6    165     7824    6    165     7824     42      981
       15     6    164     7821    6    164     7821     42      981
       16     6    164     7821    6    164     7821     42      981
     tot    100   2635   125189  100   2635   125189    661    15663
 


     bravais-lattice index     =            0
     lattice parameter (a_0)   =       1.0000  a.u.
     unit-cell volume          =     234.2976 (a.u.)^3
     number of atoms/cell      =            2
     number of atomic types    =            1
     number of electrons       =        12.00
     number of Kohn-Sham states=           10
     kinetic-energy cutoff     =     250.0000  Ry
     charge density cutoff     =    1000.0000  Ry
     convergence threshold     =      1.0E-08
     mixing beta               =       0.7000
     number of iterations used =            8  plain     mixing
     Exchange-correlation      = LDA (1100)
     EXX-fraction              =        0.00

     celldm(1)=   1.000000  celldm(2)=   0.000000  celldm(3)=   0.000000
     celldm(4)=   0.000000  celldm(5)=   0.000000  celldm(6)=   0.000000

     crystal axes: (cart. coord. in units of a_0)
               a(1) = (  5.105095 -3.239935  0.000000 )  
               a(2) = (  5.105095  3.239935  0.000000 )  
               a(3) = ( -6.496906  0.000000  7.082680 )  

     reciprocal axes: (cart. coord. in units 2 pi/a_0)
               b(1) = (  0.097941 -0.154324  0.089841 )  
               b(2) = (  0.097941  0.154324  0.089841 )  
               b(3) = (  0.000000  0.000000  0.141189 )  


     PseudoPot. # 1 for O  read from file O.BFD.upf
     Pseudo is Norm-conserving, Zval =  6.0
     Generated by new atomic code, or converted to UPF format
     Using radial grid of 2000 points,  1 beta functions with: 
                l(1) =   0

     atomic species   valence    mass     pseudopotential
        O              6.00    15.99900     O ( 1.00)

     No symmetry found
                                    s                        frac. trans.

      isym =  1     identity                                     

 cryst.   s( 1) = (     1          0          0      )    f =(  0.0000000 )
                  (     0          1          0      )       (  0.0000000 )
                  (     0          0          1      )       (  0.0000000 )

 cart.    s( 1) = (  1.0000000  0.0000000  0.0000000 )    f =(  0.0000000 )
                  (  0.0000000  1.0000000  0.0000000 )       (  0.0000000 )
                  (  0.0000000  0.0000000  1.0000000 )       (  0.0000000 )


     point group C_1 (1)    
     there are  1 classes
     the character table:

       E    
A      1.00

     the symmetry operations in each class:
     E        1

   Cartesian axes

     site n.     atom                  positions (a_0 units)
         1           O   tau(  1) = (   0.0000000   0.0000000   1.0865925  )
         2           O   tau(  2) = (   0.0000000   0.0000000  -1.0865925  )

   Crystallographic axes

     site n.     atom                  positions (cryst. coord.)
         1           O   tau(  1) = (  0.0976207  0.0976207  0.1534154  )
         2           O   tau(  2) = ( -0.0976207 -0.0976207 -0.1534154  )

     number of k points=     1  gaussian broad. (Ry)=  0.0001     ngauss = -99
                       cart. coord. in units 2pi/a_0
        k(    1) = (   0.0000000   0.0000000   0.0000000), wk =   2.0000000

                       cryst. coord.
        k(    1) = (   0.0000000   0.0000000   0.0000000), wk =   2.0000000

     G cutoff =   25.3303  ( 125189 G-vectors)     FFT grid: ( 64, 64,100)

     Largest allocated arrays     est. size (Mb)     dimensions
        Kohn-Sham Wavefunctions         0.15 Mb     (    978,  10)
        NL pseudopotentials             0.03 Mb     (    978,   2)
        Each V/rho on FFT grid          0.44 Mb     (  28672)
        Each G-vector array             0.06 Mb     (   7827)
        G-vector shells                 0.06 Mb     (   7251)
     Largest temporary arrays     est. size (Mb)     dimensions
        Auxiliary wavefunctions         0.60 Mb     (    978,  40)
        Each subspace H/S matrix        0.02 Mb     (     40,  40)
        Each <psi_i|beta_j> matrix      0.00 Mb     (      2,  10)
        Arrays for rho mixing           3.50 Mb     (  28672,   8)

     Initial potential from superposition of free atoms

     starting charge    0.95494, renormalised to   12.00000
     Starting wfc are    8 atomic +    2 random wfc

     total cpu time spent up to now is      0.40 secs

     per-process dynamical memory:     5.3 Mb

     Self-consistent Calculation

     iteration #  1     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.00E-02,  avg # of iterations =  4.0

     total cpu time spent up to now is      0.49 secs

     total energy              =     -63.49569430 Ry
     Harris-Foulkes estimate   =     -63.57355567 Ry
     estimated scf accuracy    <       0.15934912 Ry

     iteration #  2     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.33E-03,  avg # of iterations =  2.0

     total cpu time spent up to now is      0.57 secs

     total energy              =     -63.54884989 Ry
     Harris-Foulkes estimate   =     -63.62213585 Ry
     estimated scf accuracy    <       0.31335767 Ry

     iteration #  3     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.33E-03,  avg # of iterations =  1.0

     total cpu time spent up to now is      0.64 secs

     total energy              =     -63.48132260 Ry
     Harris-Foulkes estimate   =     -63.55064731 Ry
     estimated scf accuracy    <       0.14735812 Ry

     iteration #  4     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.23E-03,  avg # of iterations =  1.0

     total cpu time spent up to now is      0.71 secs

     total energy              =     -63.61598109 Ry
     Harris-Foulkes estimate   =     -63.70871979 Ry
     estimated scf accuracy    <       0.46218388 Ry

     iteration #  5     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.23E-03,  avg # of iterations =  1.0

     total cpu time spent up to now is      0.78 secs

     total energy              =     -63.54284070 Ry
     Harris-Foulkes estimate   =     -63.61751726 Ry
     estimated scf accuracy    <       0.24941144 Ry

     iteration #  6     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.23E-03,  avg # of iterations =  1.0

     total cpu time spent up to now is      0.85 secs

     total energy              =     -63.50893104 Ry
     Harris-Foulkes estimate   =     -63.54502438 Ry
     estimated scf accuracy    <       0.08003716 Ry

     iteration #  7     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      0.92 secs

     total energy              =     -63.58370146 Ry
     Harris-Foulkes estimate   =     -63.63435704 Ry
     estimated scf accuracy    <       0.34337443 Ry

     iteration #  8     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      0.99 secs

     total energy              =     -63.54656531 Ry
     Harris-Foulkes estimate   =     -63.58427686 Ry
     estimated scf accuracy    <       0.22498846 Ry

     iteration #  9     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.07 secs

     total energy              =     -63.53455889 Ry
     Harris-Foulkes estimate   =     -63.54707304 Ry
     estimated scf accuracy    <       0.13729862 Ry

     iteration # 10     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.14 secs

     total energy              =     -63.54656792 Ry
     Harris-Foulkes estimate   =     -63.54857267 Ry
     estimated scf accuracy    <       0.08851889 Ry

     iteration # 11     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.21 secs

     total energy              =     -63.54579835 Ry
     Harris-Foulkes estimate   =     -63.54657194 Ry
     estimated scf accuracy    <       0.08385967 Ry

     iteration # 12     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.28 secs

     total energy              =     -63.54047934 Ry
     Harris-Foulkes estimate   =     -63.54579898 Ry
     estimated scf accuracy    <       0.08203583 Ry

     iteration # 13     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.36 secs

     total energy              =     -63.54331221 Ry
     Harris-Foulkes estimate   =     -63.54458315 Ry
     estimated scf accuracy    <       0.13136965 Ry

     iteration # 14     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.43 secs

     total energy              =     -63.53944050 Ry
     Harris-Foulkes estimate   =     -63.54331672 Ry
     estimated scf accuracy    <       0.12437832 Ry

     iteration # 15     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.67E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.50 secs

     total energy              =     -63.54368303 Ry
     Harris-Foulkes estimate   =     -63.54450731 Ry
     estimated scf accuracy    <       0.07898338 Ry

     iteration # 16     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.58E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.57 secs

     total energy              =     -63.54114296 Ry
     Harris-Foulkes estimate   =     -63.54368371 Ry
     estimated scf accuracy    <       0.07697739 Ry

     iteration # 17     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.41E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.65 secs

     total energy              =     -63.54223108 Ry
     Harris-Foulkes estimate   =     -63.54370655 Ry
     estimated scf accuracy    <       0.12862194 Ry

     iteration # 18     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.41E-04,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.72 secs

     total energy              =     -63.54251154 Ry
     Harris-Foulkes estimate   =     -63.54252156 Ry
     estimated scf accuracy    <       0.00081532 Ry

     iteration # 19     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.79 secs

     total energy              =     -63.54253469 Ry
     Harris-Foulkes estimate   =     -63.54252652 Ry
     estimated scf accuracy    <       0.00304996 Ry

     iteration # 20     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.86 secs

     total energy              =     -63.54253214 Ry
     Harris-Foulkes estimate   =     -63.54253716 Ry
     estimated scf accuracy    <       0.00760434 Ry

     iteration # 21     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      1.93 secs

     total energy              =     -63.54245102 Ry
     Harris-Foulkes estimate   =     -63.54253254 Ry
     estimated scf accuracy    <       0.00567060 Ry

     iteration # 22     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.01 secs

     total energy              =     -63.54260388 Ry
     Harris-Foulkes estimate   =     -63.54258988 Ry
     estimated scf accuracy    <       0.03169395 Ry

     iteration # 23     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.08 secs

     total energy              =     -63.54258813 Ry
     Harris-Foulkes estimate   =     -63.54260449 Ry
     estimated scf accuracy    <       0.03738937 Ry

     iteration # 24     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.15 secs

     total energy              =     -63.54264833 Ry
     Harris-Foulkes estimate   =     -63.54258882 Ry
     estimated scf accuracy    <       0.03127086 Ry

     iteration # 25     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.22 secs

     total energy              =     -63.54247865 Ry
     Harris-Foulkes estimate   =     -63.54265876 Ry
     estimated scf accuracy    <       0.05591347 Ry

     iteration # 26     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.30 secs

     total energy              =     -63.54251186 Ry
     Harris-Foulkes estimate   =     -63.54253324 Ry
     estimated scf accuracy    <       0.00652189 Ry

     iteration # 27     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  6.79E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.37 secs

     total energy              =     -63.54252038 Ry
     Harris-Foulkes estimate   =     -63.54252041 Ry
     estimated scf accuracy    <       0.00024884 Ry

     iteration # 28     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  2.07E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.44 secs

     total energy              =     -63.54251936 Ry
     Harris-Foulkes estimate   =     -63.54252038 Ry
     estimated scf accuracy    <       0.00023316 Ry

     iteration # 29     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  1.94E-06,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.52 secs

     total energy              =     -63.54251990 Ry
     Harris-Foulkes estimate   =     -63.54251990 Ry
     estimated scf accuracy    <       0.00000108 Ry

     iteration # 30     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  9.02E-09,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.59 secs

     total energy              =     -63.54251990 Ry
     Harris-Foulkes estimate   =     -63.54251990 Ry
     estimated scf accuracy    <       0.00000031 Ry

     iteration # 31     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  2.56E-09,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.66 secs

     total energy              =     -63.54251990 Ry
     Harris-Foulkes estimate   =     -63.54251990 Ry
     estimated scf accuracy    <       0.00000006 Ry

     iteration # 32     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  5.24E-10,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.73 secs

     total energy              =     -63.54251990 Ry
     Harris-Foulkes estimate   =     -63.54251990 Ry
     estimated scf accuracy    <       0.00000038 Ry

     iteration # 33     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  5.24E-10,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.80 secs

     total energy              =     -63.54251990 Ry
     Harris-Foulkes estimate   =     -63.54251990 Ry
     estimated scf accuracy    <       0.00000018 Ry

     iteration # 34     ecut=   250.00 Ry     beta=0.70
     Davidson diagonalization with overlap
     ethr =  5.24E-10,  avg # of iterations =  1.0

     total cpu time spent up to now is      2.85 secs

     End of self-consistent calculation

          k = 0.0000 0.0000 0.0000 ( 15663 PWs)   bands (ev):

   -27.4951 -13.2187  -8.4715  -7.2159  -7.1627   0.7532   0.7539   5.5975
    11.7422  12.6551

     occupation numbers 
     2.0000   2.0000   2.0000   2.0000   2.0000   1.1323   0.8677   0.0000
     0.0000   0.0000

     the Fermi energy is     0.7536 ev

!    total energy              =     -63.54251990 Ry
     Harris-Foulkes estimate   =     -63.54251990 Ry
     estimated scf accuracy    <          4.8E-09 Ry

     The total energy is the sum of the following terms:

     one-electron contribution =     -46.03889693 Ry
     hartree contribution      =      27.23266430 Ry
     xc contribution           =     -13.47067727 Ry
     ewald contribution        =     -31.26533626 Ry
     - averaged Fock potential =       0.00000000 Ry
     + Fock energy             =       0.00000000 Ry
     smearing contrib. (-TS)   =      -0.00027375 Ry

     convergence has been achieved in  34 iterations

     Writing output data file pwscf.save

     Writing output data file pwscf.save
 
     init_run     :      0.37s CPU      0.37s WALL (       1 calls)
     electrons    :      2.30s CPU      2.45s WALL (       1 calls)

     Called by init_run:
     wfcinit      :      0.01s CPU      0.01s WALL (       1 calls)
     potinit      :      0.16s CPU      0.16s WALL (       1 calls)

     Called by electrons:
     c_bands      :      0.98s CPU      0.99s WALL (      34 calls)
     sum_band     :      0.31s CPU      0.32s WALL (      34 calls)
     v_of_rho     :      0.11s CPU      0.11s WALL (      35 calls)
     v_h          :      0.03s CPU      0.05s WALL (      35 calls)
     v_xc         :      0.08s CPU      0.06s WALL (      35 calls)
     mix_rho      :      0.13s CPU      0.13s WALL (      34 calls)

     Called by c_bands:
     init_us_2    :      0.00s CPU      0.01s WALL (      69 calls)
     cegterg      :      0.98s CPU      0.99s WALL (      34 calls)

     Called by sum_band:

     Called by *egterg:
     h_psi        :      0.90s CPU      0.92s WALL (      73 calls)
     g_psi        :      0.00s CPU      0.01s WALL (      38 calls)
     cdiaghg      :      0.04s CPU      0.02s WALL (      72 calls)
     cegterg:over :      0.01s CPU      0.01s WALL (      38 calls)
     cegterg:upda :      0.00s CPU      0.01s WALL (      38 calls)
     cegterg:last :      0.00s CPU      0.01s WALL (      34 calls)

     Called by h_psi:
     h_psi:vloc   :      0.89s CPU      0.91s WALL (      73 calls)
     h_psi:vnl    :      0.01s CPU      0.01s WALL (      73 calls)
     add_vuspsi   :      0.00s CPU      0.00s WALL (      73 calls)

     General routines
     calbec       :      0.01s CPU      0.00s WALL (      73 calls)
     cft3s        :      1.20s CPU      1.22s WALL (    1887 calls)
 
     Parallel routines
     fft_scatter  :      0.35s CPU      0.18s WALL (    1887 calls)
     EXX routines
 
     PWSCF        :     2.88s CPU time,        3.06s WALL time

 
   This run was terminated on:  13:24:47  31Oct2014            

=------------------------------------------------------------------------------=
   JOB DONE.
=------------------------------------------------------------------------------=
